File: erettsegi/kozep/2025_okt/python/szamlalas.py
with open('meres.txt') as meresek_input:
    meresek = [ int(k) for k in meresek_input.readline().strip().split(', ') ]

osszes_kerekparos = sum(max(0, k) for k in meresek)

print(f'2. Feladat: Összesen {osszes_kerekparos} kerékpárost számoltak')
print(f'3. Feladat: Óránkénti mérések:')

orankent_athaladokat_general = ((
    max(0, meresek[i]),
    max(0, meresek[i + 1]),
    max(0, meresek[i + 2]),
    max(0, meresek[i + 3]))
for i in range(0, len(meresek), 4))

orankenti_athaladok = [ sum(k) for k in orankent_athaladokat_general ]

for i in range(0, len(orankenti_athaladok)):
    print(f'{i + 6} órától {orankenti_athaladok[i]} kerékpáros')

legmagasabb_meres_i = max((i for i in range(0, len(meresek))), key = lambda i: meresek[i])
percek = 6 * 60 + 15 + (legmagasabb_meres_i * 15);

print(f'Az áthaladók maximális száma: {meresek[legmagasabb_meres_i]}, időpontja: {percek // 60}:{percek % 60}')

File: erettsegi/kozep/2025_okt/c#/Szamlalas.cs
var meresek = File.ReadAllText("meres.txt").Trim().Split(", ")
                  .Select(int.Parse)
                  .ToArray();

var osszesKerekparos = meresek.Sum(k => Math.Max(0, k));

Console.WriteLine($"2. Feladat: Összesen {osszesKerekparos} kerékpárost számoltak");
Console.WriteLine("3. Feladat: Óránkénti mérések:");

var orankentiAthaladok = meresek.Chunk(4)
                                .Select(k => k.Sum(l => Math.Max(0, l)))
                                .ToArray();

Enumerable.Range(0, orankentiAthaladok.Length).ToList()
          .ForEach(i => Console.WriteLine($"{i + 6} órától {orankentiAthaladok[i]} kerékpáros"));

var maxAthaladokI = Enumerable.Range(0, meresek.Length)
                              .MaxBy(i => meresek[i]);

Console.WriteLine($"Az áthaladók maximális száma: {meresek[maxAthaladokI]}, időpontja: {idotFormaz(maxAthaladokI)}");


string idotFormaz(int meresI) {
    var percek = 6 * 60 + 15 + (meresI * 15);

    return (percek / 60) + ":" + (percek % 60);
}
